### DISCUSSION TITLE
realy I need a NVIDIA Card

### DISCUSSION THREAD

**Discussion by rheros** [hlky/stable-diffusion-webui#800] [Q&A] [answered]
I finished the install, the launcher tell me Found no NVIDIA driver, I have a  amd RX 6800 XT, can I use this  card to run the app

**Comment by github2911** [comment]
I have the same issue as @rheros. Any guidance would be appreciated.

**Comment by EliEron** [comment]
Nvidia is the industry standard when it comes to AI, so officially speaking most AI projects only support Nvidia cards.

Some users have gotten it to work using the steps in [this](https://github.com/sd-webui/stable-diffusion-webui/discussions/371) discussion though, so I'd give that a try. I don't have an AMD card myself so I can't vouch for whether it currently works or not.

**Reply by PyrateGFXProductions** [reply]
I have gotten my RX580 to work with SD WebUI in Win11 using DirectML via this [Git, which is an installable extension for SD WebUI](https://github.com/Merramore/stable-diffusion-webui-extension-DirectML.git).

I don't think it matters but I am also using this fork of [SD WebUI](https://github.com/hgrsikghrd/stable-diffusion-webui-directml).
 
I am getting speeds of about 2s/it which is pretty fast for my 10 year old PC. However, [nod.ai/SHARK](https://github.com/nod-ai/SHARK) is faster by about 1s/it.

**Comment by cstueckrath** [comment]
You don't need nvidia
Windows: https://gitgudblog.vercel.app/posts/stable-diffusion-amd-win10 (no torch, so many things won't work (including this webui))
Linux: See my discussion here: https://github.com/sd-webui/stable-diffusion-webui/discussions/756

**Reply by rheros** [reply]
em , i'm win11~~

**Reply by cstueckrath** [reply]
1st link works with win11, too

**Reply by rheros** [reply]
https://gitgudblog.vercel.app/posts/stable-diffusion-amd-win10 
em, I can't open it,  timeout~~

**Reply by rheros** [reply]
another link else?

**Reply by cstueckrath** [reply]
the link works for me, try again?

**Reply by rheros** [reply]
em, over a day, the link not works for me yet, thanks for the great wall I think,
thank you in any case

**Reply by cstueckrath** [reply]
here is the same guide as a youtube video: https://www.youtube.com/watch?v=ngVBjeu66QI

**Reply by cstueckrath** [reply]
the [guide](https://gitgudblog.vercel.app/posts/stable-diffusion-amd-win10) is basically this:

1. Dependencies & Preparation
I. You'll need a recent Python 3 version installed on your machine. In this guide I'm using Python version 3.10.2.
II. You'll also need a [huggingface](https://huggingface.co/) account as well as an API access key from the huggingface settings, to download the latest version of the Stable Diffusion model (1.4 as of writing this blog post).
III. Make sure python and pip are defined in your PATH:
```
> python --version

Python 3.10.2
```
IV. Open your Windows Terminal or cmd (as an administrator)
V. Go to Harish Anand's Diffusers repository, clone it into a folder you've created for this little project and pip install its contents. You'll also need ftfy and scipy, so pip install those as well:

```
git clone https://github.com/harishanand95/diffusers.git

cd diffusers && git checkout dml && pip install -e .

pip install transformers ftfy scipy
```
VI. The trick that makes this work is to use the latest nightly build of Onnx. You can download it [here](https://aiinfra.visualstudio.com/PublicPackages/_artifacts/feed/ORT-Nightly/PyPI/ort-nightly-directml/overview/). Make sure the downloaded version matches your python version. In my case I have to download the file ort_nightly_directml-1.13.0.dev20220901005-cp310-cp310-win_amd64.whl since I'm on python version 3.10.2. Once you've downloaded it to your project folder do a:
`pip install ort_nightly_directml-1.13.0.dev20220901005-cp310-cp310-win_amd64.whl`
Make sure you're not using the "dot slash" (pip install ./ort_...) relative path syntax here as this is a common mistake that prevents the pip install command from finding the local file because Windows has issues with forward slashes.

VII. Now you have to generate a Onnx version of the Stable Diffusion model. To do so run
```
cd ./diffusers/examples/inference

huggingface-cli login
```
and paste in your [huggingface API](https://huggingface.co/settings/tokens) token when prompted for it.

If this step fails, you probably didn't accept the terms and conditions of the Stable Diffusion model yet so head over [to the model](https://huggingface.co/CompVis/stable-diffusion-v1-4) and do so before retrying the step above. Then run the ./save_onnx.py script

`python ./save_onnx.py`

2. Usage
There's the file called dml_onnx.py that can be used to generate your images. You can use as a reference or edit it according to your needs. At the bottom you can change the actual prompt, the seed, the output filename, the no. of steps, the guidance scale and so on.

If you want to change the default image resolution make sure that the widths and heights you choose can be divided by 8 and follow the instructions at the bottom of the dml_onnx.py file, such as re-generating a Onnx model for that new resolution.

**Reply by cstueckrath** [reply]
when this worked for you and you got your first picture with `python dml_onnx.py`
you can edit dml_onnx.py like this:

```
import random
[...]
rand = random.randint(1, 10000000000000000000)
torch.manualseed(rand)
[...]
image.save("Dml"+str(rand)+".png")
```
Then run this with `for /l %q in (0) do python dml_onnx.py`

this will use a new seed every time for the given prompt in dml_onnx.py

**Comment by rheros** [comment, marked as answer]
I get some info from web,

**Current State:**
My device : AMD RadeonRX 6800XT 
My System : Windows11 (I find that Rocm not support Windows , GOD!!! )

**TODO:**
1. I need to Install a ubuntu on my pc
2. install rocm
3. follow [this](https://github.com/sd-webui/stable-diffusion-webui/discussions/371)

I will have a try tomorrow , See you

**Reply by rheros** [reply]
## System prepare:
1. Install Ubuntu 20.04    https://ubuntu.com/download/alternative-downloads
2. lower kernel    https://blog.csdn.net/weixin_46123670/article/details/125777787
3. reboot
## Graphic preprare:
1. make sure system setting can see Graphics AMD® Radeon rx 6800 xt
    * if can't see  install graphics driver  https://www.amd.com/zh-hans/support/graphics/amd-radeon-6000-series/amd-radeon-6800-series/amd-radeon-rx-6800-xt   this is a installer ,make driven install easy
    * other version can search from here https://www.amd.com/zh-hans/support
## install rocm    
(doc here)  https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.2.3/page/How_to_Install_ROCm.html
    *  sudo apt-get update
    *  wget https://repo.radeon.com/amdgpu-install/22.20.3/ubuntu/focal/amdgpu-install_22.20.50203-1_all.deb 
    *  sudo apt-get install ./amdgpu-install_22.20.50203-1_all.deb
    * sudo amdgpu-install --usecase=rocm
    * sudo amdgpu-install --usecase=dkms
## check rocm install success? use shell cmd one of blow
  rocm-smi
  /opt/rocm/bin/rocminfo
  * if display graphics info ,congratulations you success
  
## install conda
site https://docs.conda.io/en/latest/miniconda.html#linux-installers

## install stable diffusion follow this 
 * https://sd-webui.github.io/stable-diffusion-webui/docs/2.linux-installation.html  this is the recommend way
 * I git clone the repo, and run the webui.sh under the stable-diffusion-webui folder
 * remember download the sd-v1-4.ckpt mode and mv it in the stable-diffusion-webui ,webui.sh need it

## install pythorch in ldm env，otherwise the webui.sh will tell you ,you don't have cuda ,
pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/rocm5.2/   

## downlond  sd-v1-4.ckpt  mode 
https://huggingface.co/CompVis/stable-diffusion-v-1-4-original
## no module named cv2
1. enter the conda env ldm and install opencv use shell
     * conda  activate ldm
     * pip install opencv-contrib-python

**Reply by rheros** [reply]
* don't run webui.sh under sudo, this will make some folder user and  group to root
* my RX6800 XT can display use rocm-smi, but torch show  no cuda , change the the version in environment.yaml  pytorch==1.12.1

**Reply by cstueckrath** [reply]
interesting - so you didn't use the ROCm pytorch?
`pip3 install --upgrade torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/rocm5.1.1`

**Reply by rheros** [reply]
em,I forget witch step to install it
it should be installed and I surely have installed it

**Reply by rheros** [reply]
yes, I rm my ldm env, reinstall stable diffusion, webui.sh show err i don't have a nvidia card,  yeah, I need to install 
pip3 install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/rocm5.2/   
thanks a lot  bro
